src/main/java/ui/BoardAutoCreator.java

```diff
@@ -84,8 +84,11 @@ public Menu generateBoardAutoCreateMenu() {
         return autoCreate;
     }
 
-    private void saveBoardAfterUserConfirmation(String boardName) {
-        if (isSaveBoardDialogResponsePositive(boardName)) ui.getMenuControl().saveBoardAs();
+    private void saveBoardAfterUserConfirmationIfNeeded(String boardName) {
+        boolean isNeeded = !prefs.getLastOpenBoardPanelInfos().isPresent()
+                        || !prefs.getLastOpenBoardPanelInfos().get().equals(panelControl.getCurrentPanelInfos());
```

Comment:
I think a more accurate name is `isOpenBoardDifferentFromSavedBoard` or something like that, because that's what the condition seems to be checking.